src/main/scala/http4s/extend/util/ThrowableModule.scala

```diff
@@ -0,0 +1,35 @@
+package http4s.extend.util
+
+import http4s.extend.Algebra.ThrowableCompleteMessage
+
+import scala.annotation.tailrec
+
+object ThrowableModule {
+
+  private val separator = "\n\rcaused by "
+
+  def throwableOfMessage: ThrowableCompleteMessage => Throwable =
+    xs => foldedThrowable(xs.message.split(separator).toSeq map (new ThrowableCompleteMessage(_)))
+
+  def foldedThrowable: Seq[ThrowableCompleteMessage] => Throwable =
```

Comment:
Maybe make private those functions that are meant to be used within the module

Reply:
At the beginning they were, but then I changed my mind. They might be helpful in case you want to build a Throwable instead of converting one. Let's say you create a list of messages from strings. You can build a throwable from that. I'm thinking to tests in particular.